From 697bf748e6969543073b7eb7b2043b8d673c6b2c Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@nxp.com>
Date: Fri, 23 Feb 2018 14:13:20 +0200
Subject: [PATCH 4100/5242] MLK-18691: ASoC: imx-xtor: don't add ASRC audio
 routes if node is missing

commit  4911bb5938d3a3e58aa9b7adf66ead56b92bce9a from
https://source.codeaurora.org/external/imx/linux-imx.git

ASRC audio routes introduce aditional log noise if ASRC
OF node is missing in DTS, so add them if ASRC node is present only.

Signed-off-by: Viorel Suman <viorel.suman@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-xtor.c |   11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/sound/soc/fsl/imx-xtor.c b/sound/soc/fsl/imx-xtor.c
index 3b59553..39e3a8e 100644
--- a/sound/soc/fsl/imx-xtor.c
+++ b/sound/soc/fsl/imx-xtor.c
@@ -148,14 +148,12 @@ static int be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 }
 
 static const struct snd_soc_dapm_route audio_map[] = {
-	/* Line out jack */
+	{"Playback",  NULL, "CPU-Playback"},
+	{"CPU-Capture",  NULL, "Capture"},
 	{"CPU-Playback",  NULL, "ASRC-Playback"},
-	{"Playback",  NULL, "CPU-Playback"},/* dai route for be and fe */
 	{"ASRC-Capture",  NULL, "CPU-Capture"},
-	{"CPU-Capture",  NULL, "Capture"},
 };
 
-
 static struct snd_soc_ops imx_xtor_ops = {
 	.startup = imx_xtor_startup,
 	.shutdown  = imx_xtor_shutdown,
@@ -224,6 +222,8 @@ static int imx_xtor_probe(struct platform_device *pdev)
 	data->dai[0].capture_only = false;
 	data->card.num_links = 1;
 	data->card.dai_link = data->dai;
+	data->card.dapm_routes = audio_map;
+	data->card.num_dapm_routes = 2;
 
 	/*if there is no asrc controller, we only enable one device*/
 	if (asrc_pdev) {
@@ -250,6 +250,7 @@ static int imx_xtor_probe(struct platform_device *pdev)
 		data->dai[2].be_hw_params_fixup = be_hw_params_fixup,
 		data->card.num_links = 3;
 		data->card.dai_link = &data->dai[0];
+		data->card.num_dapm_routes += 2;
 
 		ret = of_property_read_u32(asrc_np, "fsl,asrc-rate",
 						&data->asrc_rate);
@@ -272,8 +273,6 @@ static int imx_xtor_probe(struct platform_device *pdev)
 			data->asrc_format = SNDRV_PCM_FORMAT_S16_LE;
 	}
 
-	data->card.dapm_routes = audio_map,
-	data->card.num_dapm_routes = ARRAY_SIZE(audio_map),
 	data->card.dev = &pdev->dev;
 	data->card.owner = THIS_MODULE;
 	ret = snd_soc_of_parse_card_name(&data->card, "model");
-- 
1.7.9.5

